### PR TITLE
feat: Enable Hubble Relay in Cilium deployment via CAAPH

### PR DESCRIPTION
**Pull request by dlipovetsky**
**What problem does this PR solve?**:
Relay is required for network visibility of a cluster as a whole. See https://docs.cilium.io/en/stable/gettingstarted/hubble_intro/.

**Which issue(s) this PR fixes**:
Fixes #

**How Has This Been Tested?**:

**Special notes for your reviewer**:

### CODE REVIEW

**Review by faiq — APPROVED** [review]

**Review by jimmidyson — APPROVED** [review]

File: charts/cluster-api-runtime-extensions-nutanix/templates/cni/cilium/manifests/helm-addon-installation.yaml [hunk 1]
```diff
@@ -12,6 +12,7 @@ data:
       chainingMode: portmap
       exclusive: false
     hubble:
+      relay: enabled
```
Comment (jimmidyson) [inline charts/cluster-api-runtime-extensions-nutanix/templates/cni/cilium/manifests/helm-addon-installation.yaml:15]: Ran locally and cilium install failed with (taken from Cilium HCP):

```
values don't meet the specifications of the schema(s) in the following chart(s):
cilium:
- hubble.relay: Invalid type. Expected: object, given: string
```
```suggestion
      relay:
        enabled: true
```

**Review by jimmidyson — CHANGES_REQUESTED** [review]

**Comment by jimmidyson** [comment]
@dlipovetsky Please can you add a test to e2e to ensure the necessary hubble components are running? This will also help us know what to expect in-cluster when Hubble is enabled.

**Comment by jimmidyson** [comment]
@dlipovetsky Force pushed after editing last commit message to pass gitlint.

**Comment by jimmidyson** [comment]
Added e2e test in 56a583c7187cdb7390e233966a90fb3823d517c6 (while also taking the opportunity to tidy up "kube-system" consts).

**Review by jimmidyson — APPROVED** [review]

**Review by mhrabovcin — APPROVED** [review]

**Pull request merged**